libobs/obs-output.c

```diff
@@ -1340,6 +1397,24 @@ void obs_output_set_audio_conversion(
 	output->audio_conversion_set = true;
 }
 
+// Enable or disable broadcast performance metrics reporting
+void obs_output_enable_bpm(obs_output_t *output, bool bpm_enable)
+{
+	if (!obs_output_valid(output, "obs_output_enable_bpm"))
+		return;
+	if (!log_flag_service(output, __FUNCTION__))
+		return;
+	if (active(output))
+		return;
+
+	os_atomic_set_bool(&output->info.enable_bpm, bpm_enable);
```

Comment:
You should not modify an info structure.

Reply:
Thank you, I understand now what the info structure is meant to do now. I've reworked this, hopefully it's fine now.